packages/netlify-cms-core/src/actions/auth.ts

```diff
@@ -115,3 +118,11 @@ export function logoutUser() {
     });
   };
 }
+
+export type AuthAction = ReturnType<
```

Comment:
Defining action type like this is much more type safe than
https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-core/src/types/redux.ts#L402-L413